src/pkg/services/m365/api/groups.go

```diff
@@ -46,13 +46,26 @@ func (c Teams) GetAll(
 		return nil, err
 	}
 
-	return getGroups(ctx, true, errs, service)
+	return getGroups(ctx, func(ctx context.Context, g models.Groupable) bool { return true }, errs, service)
+}
+
+// GetTeams retrieves all Teams.
+func (c Groups) GetTeams(
+	ctx context.Context,
+	errs *fault.Bus,
+) ([]models.Groupable, error) {
+	service, err := c.Service()
+	if err != nil {
+		return nil, err
+	}
+
+	return getGroups(ctx, FetchOnlyTeams, errs, service)
 }
 
 // GetAll retrieves all groups.
 func getGroups(
 	ctx context.Context,
-	getOnlyTeams bool,
+	filterGroupsData func(ctx context.Context, g models.Groupable) bool,
```

Comment:
Hmm, this isn't my preferred implementation.  Ad-hoc funcs for filtering are genrally a more complicated implementation of a standalone transformer func.  I think a more simple implementation is to split this out to the following funcs.  These are each more simple to grok, more simple to test, and can be exported if we need their behavior elsewhere.

```go
// gets all groups, does no filtering

func (c Client) getGroups(ctx, errs) ([]models.Groupable, error) {...}

// callers can pass the results of getGroups() here.

func removeTeams([]models.Groupable) []models.Groupable {...}

func removeNonTeams([]models.Groupable) []models.Groupable {...}
```

Reply:
Got it. Something similar to what I implemented initially.